app/src/main/java/ai/elimu/vitabu/ui/storybook/StoryBookActivity.kt

```diff
@@ -47,6 +51,34 @@ class StoryBookActivity : AppCompatActivity() {
 
         val zoomOutPageTransformer = ZoomOutPageTransformer()
         binding.viewPager.setPageTransformer(true, zoomOutPageTransformer)
+        binding.viewPager.setOnPageChangeListener(object : ViewPager.OnPageChangeListener {
+
+            override fun onPageScrolled(
+                position: Int,
+                positionOffset: Float,
+                positionOffsetPixels: Int
+            ) {
+            }
+
+            override fun onPageSelected(position: Int) {
```

Comment:
@tuancoltech This code triggers immediately when the student reaches the last page, which means that the time spent reading the last page won't be included. Technically a book completion is when the student finishes reading this last page, so would it be possible to move this code into the Activity's `onStop()` or `onDestroy()` instead? If it's possible to get the current page position there.